src/PhpSpreadsheet/Worksheet/Worksheet.php

```diff
@@ -2922,23 +2922,30 @@ public function fromArray(array $source, $nullValue = null, $startCell = 'A1', $
         [$startColumn, $startRow] = Coordinate::coordinateFromString($startCell);
 
         // Loop through $source
-        foreach ($source as $rowData) {
-            $currentColumn = $startColumn;
-            foreach ($rowData as $cellValue) {
-                if ($strictNullComparison) {
+        if ($strictNullComparison) {
+            foreach ($source as $rowData) {
+                $currentColumn = $startColumn;
+                foreach ($rowData as $cellValue) {
                     if ($cellValue !== $nullValue) {
                         // Set cell value
                         $this->getCell($currentColumn . $startRow)->setValue($cellValue);
```

Comment:
I've not run any timing tests on your code yet; but when it comes to micro-optimisations, then interpolation is more efficient than the concatenation that we use here:
```php
$this->getCell("{$currentColumn}{$startRow}")->setValue($cellValue);
```

Try running
```php
$callStartTime = microtime(true);

for ($row = 1; $row <= 65535; ++$row) {
    for ($column = 'A'; $column !== 'IV'; ++$column) {
        $cellAddress = $column . $row;
    }
}

$callEndTime = microtime(true);
$concatenationCallTime = $callEndTime - $callStartTime;

echo PHP_EOL;
echo 'Call time for concatenation was ' , sprintf('%.4f', $concatenationCallTime) , ' seconds' , PHP_EOL;


$callStartTime = microtime(true);

for ($row = 1; $row <= 65535; ++$row) {
    for ($column = 'A'; $column !== 'IV'; ++$column) {
        $cellAddress = "{$column}{$row}";
    }
}

$callEndTime = microtime(true);
$interpolationCallTime = $callEndTime - $callStartTime;

echo PHP_EOL;
echo 'Call time for interpolation was ' , sprintf('%.4f', $interpolationCallTime) , ' seconds' , PHP_EOL;
```
to see the difference

And I'm of the growing opinion that interpolation is actually more readable

Reply:
Ticket amazingly has no merge conflicts after all this time. I have run Mark's test program, and it now seems to favor concatenation over interpolation. So I will let this proceed as-is, assuming it still passes all tests, of course.